### PR TITLE
Blog app: add authorization rules ⚒️

### PR DESCRIPTION
**Pull request by ritikarawat220**
### Project requirements  💭

- Install **`CanCanCan`** in your project.
- Add a `role` column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin `role` (column role has value `"admin"`). Use CanCanCan for this authorization.
- For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin `role` (column role has value `"admin"`). Use CanCanCan for this authorization.
- For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    user ||= current_user # guest user (not logged in)
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :read, :all
+      can :create, [Post, Comment]
+      can :update, [Post, Comment], author_id: user.id
+      can :destroy, Post, author_id: user.id
+      can :destroy, Comment, author_id: user.id
+    end
```
Comment (elisha2kyakpo1) [inline app/models/ability.rb:15]: - [ ] Great job adding the ability class, unfortunately, I couldn't create a post using a user or admin role. Kindly fix the issue before submitting the PR for review. To test this project a post, comment and a like should be create.

authorization

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @ritikarawat220, @jeanbuhendwa, @MMhaidari, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file👏
👏 The app works as expected.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/ability.rb (hunk 1 above)
Comment (elisha2kyakpo1) [reply app/models/ability.rb:15]: Sorry I approved the PR instead of requested review, the dashboard will show changes required.

Comment (muneebulrehman) [reply app/models/ability.rb:15]: It seems the issue of creating the post on behalf of logged in user still persists. Take a look at the below screenshot of the console. ⬇️
image

Here is the screen recording of the application. ⬇️


https://github.com/ritikarawat220/Blog-App/assets/33268927/00adc4a6-11a3-47cc-92d9-75da52c06046

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @ritikarawat220 , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Good follow-up with the previous reviewer.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] I would recommend implementing the `destroy` action in the `PostsController` also just like you did in the `CommentsController`. Since, the previous reviewer didn't raise this as required, I am marking this as optional. However, this is part of the project requirements and hence, I would strongly recommend you implement it.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @ritikarawat220, @MMhaidari and @jeanbuhendwa,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  No Linters errors 🔥
- [ ]  Great job working on the rules💯
- [ ]  Followed GitFlow ⭐🚀
- [ ] Descriptive PR 🚀
- [ ] All functionality work as expected 💯💯


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**